src/components/shared/swipeable-modal/SwipeableModal.tsx

```diff
@@ -215,26 +102,90 @@ class SwipeableModal extends React.Component<
       position: positionTarget,
       targets: animateObject,
       update: () => {
-        this.setState({
-          lastPosition: animateObject.position,
-          opacity: 1 - this.swipeRatio(animateObject.position, swipableWidth),
-          swiping: false,
-          translateX: `${animateObject.position}px`,
-        });
+        setLastPosition(animateObject.position);
+        setOpacity(1 - swipeRatio(animateObject.position, swipeableWidth));
+        setTranslateX(`${animateObject.position}px`);
       },
     });
-  }
+  };
 
-  private swipeRatio(position: number, width: number) {
-    const initialPosition = this.initialDivPosition;
-    const diff = position - initialPosition;
+  return {
+    frontDivRef,
+    opacity,
+    translateX,
+  };
+}
 
-    return diff / width;
-  }
+function useLockedBackgroundContent() {
+  React.useEffect(() => {
+    document.body.style.overflow = "hidden";
+    return () => {
+      document.body.style.overflow = "";
+    };
+  }, []);
+}
+
+function SwipeableModalContent({
+  title,
+  closeButton,
+  onClose,
+  className,
+  children,
+}: ModalContentProps) {
+  const showHeader = !!title || closeButton;
+  const { frontDivRef, opacity, translateX } = useSwipeToClose(onClose);
+
+  useLockedBackgroundContent();
+
+  return (
+    <>
+      <Overlay opacity={opacity} onClick={onClose} />
+
+      <div
+        ref={frontDivRef}
+        role="dialog"
+        aria-modal="true"
+        aria-labelledby="modal-title"
+        className={classNames("modal", className)}
+        style={{
+          opacity,
+          transform: `translate(${translateX}, -50%)`,
+        }}
+      >
+        {showHeader && (
+          <div className="modal__header">
+            {title && (
+              <span id="modal-title" className="modal__header__title">
+                {title}
+              </span>
+            )}
 
-  private onAction() {
-    this.close();
+            {closeButton && (
+              <IconButton
+                className="modal__header__close-button"
+                iconName="close"
+                onClick={onClose}
+              />
+            )}
+          </div>
+        )}
+
+        {children}
+      </div>
+    </>
+  );
+}
+
+function SwipeableModal({ open, ...contentProps }: IModalProps) {
+  if (!open) {
+    return null;
   }
+
+  return (
+    <Portal>
+      <SwipeableModalContent {...contentProps} />
```

Comment:
Note: the extraction of the content into it's own component was important so that the content's state and effects are only initially set when opened and destroyed when closed.